be/src/olap/rowset/beta_rowset_reader.cpp

```diff
@@ -111,9 +112,10 @@ Status BetaRowsetReader::init(RowsetReaderContext* read_context) {
     RowwiseIterator* final_iterator;
     if (config::enable_storage_vectorization && read_context->is_vec) {
         if (read_context->need_ordered_result &&
```

Comment:
Rethink the `read_orderby_key` and `need_ordered_result`, seems we only need one

Reply:
removed read_context->read_orderby_key and optimized the new_merge_iterator/new_union_iterator as we discussed.